galley/pkg/config/source/kube/rt/origin.go

```diff
@@ -48,6 +48,10 @@ func (o *Origin) FriendlyName() string {
 	return fmt.Sprintf("%s %s", o.Kind, o.FullName.String())
 }
 
+func (o *Origin) Comparator() string {
+	return o.Kind + o.FullName.Name.String() + o.FullName.Namespace.String()
```

Comment:
nit: this is not unique without delimiters

Reply:
Actually kind is also not distinct (there are two Gateway's). So if we expect them to be unique we will need more info. If its just for sorting its probably less important?

Reply:
Just for sorting.  trying to avoid 3x calls to fmt.Sprintf times n*log(n) for the sort operation.

Reply:
Having a delimiter would be much better. Kind should be fine, as they are all within Istio resource scope.

Reply:
@therealmitchconnors can you add a delimiter just to be 100% safe and I will approve. Thanks!

Reply:
ping?